kernelci/runtime/__init__.py

```diff
@@ -3,44 +3,68 @@
 # Copyright (C) 2019, 2021-2023 Collabora Limited
 # Author: Guillaume Tucker <guillaume.tucker@collabora.com>
 
+"""KernelCI Runtime environment base class definition"""
+
+import abc
 import importlib
 import json
 import os
+import yaml
+
+from jinja2 import Environment, FileSystemLoader
+
 
+def add_kci_raise(jinja2_env):
+    """Add a kci_raise function to use in templates
 
-def add_kci_raise(env):
+    This adds a `kci_raise` function to a given Jinja2 environment `jinja2_env`
+    so it can be used to raise exceptions from template files when for example
+    some template parameters are not valid.
+    """
     def template_exception(msg):
         raise Exception(msg)
```

Comment:
It's not part of this PR, but maybe it'd be good to use specialized exception class here like `TemplateError` or `TemplateParameterError` instead of a generic `Exception`. This could improve further exceptions handling.

Reply:
Yes, error handling needs to be revisited throughout the code base in general.  In this case, we might need to define several handlers or add a variable with the type of exception to raise.  Also there might be something more suitable in recent versions of Jinja2.

Reply:
Maybe we could make it a follow-up?

Reply:
#1797